homeassistant/components/nws/weather.py

```diff
@@ -92,15 +100,27 @@ async def async_setup_entry(
     ):
         entity_registry.async_remove(entity_id)
 
-    async_add_entities([NWSWeather(entry.data, nws_data)], False)
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        "get_forecasts_extra",
+        {vol.Required("type"): vol.In(("hourly", "twice_daily"))},
+        "async_get_forecasts_extra_service",
+        supports_response=SupportsResponse.ONLY,
+    )
 
+    async_add_entities([NWSWeather(entry.data, nws_data)], False)
 
-if TYPE_CHECKING:
 
-    class NWSForecast(Forecast):
-        """Forecast with extra fields needed for NWS."""
+class ExtraForecast(TypedDict, total=False):
+    """Forecast extra fields from  NWS."""
 
-        detailed_description: str | None
+    # common attributes
+    datetime: Required[str]
+    is_daytime: bool | None
+    # extra attributes
+    detailed_description: str | None
+    dewpoint: float | None
```

Comment:
Dewpoint is included in the normal forecast but maybe you want it here too?

Reply:
I did not notice that this was added to the regular weather platform. We should remove it from this PR and add it to the normal forecast in a separate PR.  There was a feature request to add a different value, so I will add here to make sure that we return something for both weather types in the forecast.

Reply:
dewpoint removed in https://github.com/home-assistant/core/pull/117254/commits/7e377cd022d6e7874bfdc78f20ddeabd06e1eab4, short_description added in same.